docs/content/docs/data-structures/map.md

```diff
@@ -1,10 +1,449 @@
+const button = document.createElement('button');
+button.textContent = "Randomize!";
+const label = document.createElement('label');
+    
+button.addEventListener('click', () =>
+    map.set(dataKey, Math.random())
+);
+
+// Get the current value of the shared data to update the view whenever it changes.
+const updateLabel = (changed, local) => {
+    const value = map.get(dataKey) || 0;
+    label.textContent = `${value} from ${local ? "me" : "someone else"}`;
+    label.style.color = changed?.previousValue > value ? "red" : "green";
+};
+updateLabel(undefined, false);
+    // Use the changed event to trigger the rerender whenever the value changes.
+map.on('valueChanged', updateLabel);
+```
+Now, with the changes in `updateLabel`, the view will appropriately re-render to say if the value was last updated by the current user or by someone else. It will also compare the current value to the last one, and if the value has increased, it will set the text color to green. Otherwise, it will be red.
+
+## Usage guides
+
+`SharedMap` supports storing
+- primitives (strings, numbers, booleans, etc.)
+- [objects](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object) 
+- shared objects (DDSes, data objects)
+
+Let's walk through each of these different types and look at best practices on how to use them with the `SharedMap` APIs.
+
+### Storing primitives
+
+As you saw in the examples above, storing and fetching primitve values from a `SharedMap` are as simple as calling the `set`/`get` functions on the same key. The common paradigm to follow is:
+1. Create a listener callback for updating the app state using `get` to fetch the latest data. This gets triggered on a `valueChanged` event
+2. Adding some app functionality, whether that is through user action or otherwise, to call `set` with new data
+3. Observe the `set` call trigger the `valueChanged` event for local and remote clients and see all of their local app states get updated
+
+Whichever client was the last to fire a `set` call will have its value be the current value on the key.
+
+### Storing objects
+
+When storing objects in a `SharedMap`, the concepts for applying and fetching updates remain the same as when storing primitives. However, one thing to note is that all updates in `SharedMap` are merged using a last writer wins (LWW) strategy. This means that if multiple clients are writing values to the same key, whoever made the last update will "win" and overwrite the others. While this is fine for primitives, please be careful when using this to store objects if you are looking for different fields within the object to be modularly modified.
+
+{{< callout warning >}}
+
+If your application requires different fields in an object to be edited simultaneously by multiple people, do NOT store the entire object under one key. Instead, store each field under its own separate key such that each one of them can be updated separately.
+
+{{< /callout >}}
```

Comment:
I think the third tip in that section already covers this point now. However, having an additional warning here might be good to reinforce this concept for the developer since this section is where they'll actively be reading about how to store objects, and they may have missed the tip at the beginning